### PR TITLE
[Merged by Bors] - chore: use FunLike.coe as coercion for OrderIso and RelEmbedding

### PR DESCRIPTION
**Pull request by ChrisHughes24**
The changes I made were.

Use `FunLike.coe` instead of the previous definition for the coercion from `RelEmbedding` To functions and `OrderIso` to functions. The previous definition was
```lean
instance : CoeFun (r ↪r s) fun _ => α → β :=
--   ⟨fun o => o.toEmbedding⟩
```
This does not display nicely.

I also restored the `simp` attributes on a few lemmas that had their `simp` attributes removed during the port. Eventually
we might want a `RelEmbeddingLike` class, but this PR does not implement that.

I also added a few lemmas that proved that coercions to function commute with `RelEmbedding.toRelHom` or similar.

The other changes are just fixing the build. One strange issue is that the lemma `Finset.mapEmbedding_apply` seems to be harder to use, it has to be used with `rw` instead of `simp`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
@ChrisHughes24, would you be able to write an explanation of what you're doing, and why, in the PR description? (Or linking to zulip as appropriate?)

File: Mathlib/Order/RelIso/Basic.lean [hunk 1]
```diff
@@ -575,16 +588,31 @@ theorem toEquiv_injective : Injective (toEquiv : r ≃r s → α ≃ β)
 instance : CoeOut (r ≃r s) (r ↪r s) :=
   ⟨toRelEmbedding⟩
 
--- see Note [function coercion]
-instance : CoeFun (r ≃r s) fun _ => α → β :=
-  ⟨fun f => f⟩
```
Comment (eric-wieser) [inline Mathlib/Order/RelIso/Basic.lean:580]: Probably should be `porting note: removed`

File: Mathlib/Order/RelIso/Basic.lean [hunk 2]
```diff
@@ -575,16 +588,31 @@ theorem toEquiv_injective : Injective (toEquiv : r ≃r s → α ≃ β)
 instance : CoeOut (r ≃r s) (r ↪r s) :=
   ⟨toRelEmbedding⟩
```
Comment (ChrisHughes24) [inline Mathlib/Order/RelIso/Basic.lean:590]:
```suggestion

-- Porting note: removed
```

File: Mathlib/Order/RelIso/Basic.lean (hunk 1 above)
Comment (ChrisHughes24) [reply Mathlib/Order/RelIso/Basic.lean:580]: Okay, I tried and failed to fix this from my phone. I'll have a proper look on Monday.

Comment (ChrisHughes24) [reply Mathlib/Order/RelIso/Basic.lean:580]: I've fixed this note now

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**